pmd-java/src/main/java/net/sourceforge/pmd/lang/java/symbols/internal/asm/ClassStub.java

```diff
@@ -315,9 +310,9 @@ public List<JTypeVar> getTypeParameters() {
     }
 
     @Override
-    public boolean isGeneric() {
+    public int getTypeParameterCount() {
```

Comment:
We implement getTypeParameterCount() explicitly for ClassStub (the default impl would call signature.getTypeParameters().size() which would trigger the full parsing of the signature).

Note: the default impl of isGeneric() is good enough and uses just the type parameter count. So, no need to override it here.